java/code/src/com/redhat/rhn/frontend/xmlrpc/contentmgmt/ContentManagementHandler.java

```diff
@@ -645,6 +651,39 @@ public ContentFilter createFilter(User loggedInUser, String name, String rule, S
         }
     }
 
+    /**
+     * Create new {@link ContentFilter}s for all AppStream modules with default streams
+     *
+     * @param loggedInUser the logged in user
+     * @param prefix the filter name prefix
+     * @param channelLabel label of the modular channel
+     * @param projectLabel label of the Content Lifecycle Project
+     * @throws EntityExistsFaultException when Filter already exist
+     * @return List of created and successfully attached Filter
+     */
+    public List<ContentFilter> createAppStreamFilters(User loggedInUser, String prefix,
+            String channelLabel, String projectLabel) throws ModulemdApiException {
+        ensureOrgAdmin(loggedInUser);
+
+        Channel channel = ChannelManager.lookupByLabelAndUser(channelLabel, loggedInUser);
+
+        try {
+            List<ContentFilter> createdFilters = filterTemplateManager.createAppStreamFilters(
+                    prefix, channel, loggedInUser);
+
+            List<ContentFilter> attachedFilters = new ArrayList<>();
+
+            for (ContentFilter createdFilter : createdFilters) {
+                attachedFilters.add(contentManager.attachFilter(projectLabel, createdFilter.getId(), loggedInUser));
+            }
+
+            return attachedFilters;
+        }
+        catch (EntityExistsException e) {
+            throw new EntityExistsFaultException(e);
+        }
```

Comment:
Same as above, here we need another catch block to catch the `EntityNotExistsException` (thrown from `attachFilter` if the project label does not exist) and throw an `EntityNotExistsFaultException`

Reply:
I combined `LookupException` and `EntityNotExistsException` in the same catch block, not sure if that style is ok or if you prefer a separate catch block per exception.